### ISSUE TITLE
Lost Password

### ISSUE THREAD

**Issue by Richard399** [AzuraCast/AzuraCast#2963]
Hi,

I have forgotten the password to the dashboard.

When using Lost Password, it takes me to your site showing you how to reset the password using SSH.

Which is fine IF you know how to use SHH?

Can this not be a more simplified approach where you get an email which provides a string to change you password automatically

Hope that makes sense :-)

Regards

Richard

**Comment by Vaalyn** [comment]
For user accounts with 2FA disabled this should be doable, although there is currently no email integration in AzuraCast (if I'm not mistaken) so that would need to be set up first when implementing this.

For accounts with 2FA enabled the workflow for this should be discussed first. There are multiple ways for this to work with 2FA: 

- require someone to go to the CLI to remove the 2FA from the account and then resetting it via "Lost Password"
- requesting a 2FA token when requesting a password reset email via "Lost Password"
  - requesting a 2FA token again when entering a new password
  - not requesting a 2FA token when entering a new password
- not requesting a 2FA token when requesting a password reset email via "Lost Password"
  - requesting a 2FA token when entering a new password

Each flow has it's drawbacks. Either convenience or security so we should decide which one would be acceptable from both perspectives.

**Comment by SlvrEagle23** [comment]
Correct, the only reason we don't implement an easier "forgot password" strategy is because it would require setting up e-mail delivery on each installation, and we've long considered that too troublesome to implement (and too error-prone) to be worth it for the one or two functions it would be used for in the system.
